src/Data/Functor/Contravariant/Conclude.hs

```diff
@@ -1,9 +1,10 @@
 {-# LANGUAGE CPP           #-}
+{-# LANGUAGE Safe          #-}
```

Comment:
In order to do this in a way that's compatible with really old GHCs, use this instead of `{-# LANGUAGE Safe #-}`:

```hs
#if __GLASGOW_HASKELL__ >= 704
{-# LANGUAGE Safe #-}
#elif __GLASGOW_HASKELL__ >= 702
{-# LANGUAGE Trustworthy #-}
#endif
```

There are two reasons for this:

1. Neither `Safe` nor `Trustworthy` existed in GHC 7.0.
2. Although `Safe` does exist in GHC 7.2, it's [quite buggy](https://github.com/haskell/cabal/issues/5010). As a result, we make an effort to [avoid using it in 7.2](https://github.com/ekmett/semigroupoids/issues/69).

Similar comments apply to the other `{-# LANGUAGE Safe #-}` additions elsewhere in this PR.